fake.go

```diff
@@ -519,6 +521,177 @@ func (fake *Fake) Dump() string {
 	return buf.String()
 }
 
+// extractComment removes comment section from a given string and returns it together with comment string.
+// If a comment wasn't found, an empty comment string will be returned.
+func extractComment(s string) (string, string) {
+	if !strings.Contains(s, "comment") {
+		return s, ""
+	}
+	// find comment section with colon
+	reg := regexp.MustCompile(`comment.*".*" ?;?`)
+	// extract comment section and leave just the string
+	comment := reg.FindStringSubmatch(s)[0]
+	commentReg := regexp.MustCompile(`"[^"]+"`)
+	comment = commentReg.FindStringSubmatch(comment)[0]
+	comment = strings.Trim(comment, "\"")
+	// remove comment section from the initial string
+	res := reg.ReplaceAllString(s, "${1}")
+	return strings.TrimSpace(res), comment
+}
+
+// ParseNFTablesDump can parse a dump for a given nft instance.
+// It expects one table declaration and the same nftables family in all rules based on a given nft.nftContext.
+// It lacks some error handling, and the best way to verify that everything important was properly parsed is to
+// compare given data with nft.Dump() output.
+func ParseNFTablesDump(nft *Fake, data string) (err error) {
+	lines := strings.Split(strings.Trim(data, "\n"), "\n")
+	var i int
+	var line string
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("line %v: %w", i+1, err)
+		}
+	}()
+	for i, line = range lines {
+		line = strings.TrimSpace(line)
+		if line == "" || line[0] == '#' {
+			continue
+		}
+		words := strings.Split(line, " ")
+		if words[0] != "add" {
+			return fmt.Errorf("only supports add operation, got %s", words[0])
+		}
+		if words[2] != string(nft.nftContext.family) {
+			return fmt.Errorf("wrong ip family %s, expected %s", words[2], string(nft.nftContext.family))
+		}
+		if words[3] != nft.nftContext.table {
+			return fmt.Errorf("wrong table name %s, expected %s", words[3], nft.nftContext.table)
+		}
+		if words[1] != "table" && nft.Table == nil {
+			return fmt.Errorf("no table was added, can't add %s", words[1])
+		}
+		switch words[1] {
+		case "table":
+			if nft.Table != nil {
+				return fmt.Errorf("only 1 table should be declared")
```

Comment:
`add` is idempotent. You're allowed to re-add the same table twice. You should compare the semantics here against `run()`.

Actually... it would probably be simpler to have this function just create `Table`, `Chain`, `Set`, etc objects, add them all to a `Transaction`, and then `Run` it, and let `Run` deal with all the error checking. (eg, your code will crash if someone tries to add a rule to a chain that doesn't exist)

Given that idea, and the comment above about "ought to be able to parse anything that fake.Dump might output", maybe there should be per-object-type parsing functions in `objects.go`, which would be parallel to the `writeOperation` methods and then it would be obvious then when we added new stuff we needed to update both methods.

Reply:
> maybe there should be per-object-type parsing functions in `objects.go`

(that could come later...)